src/Services/ElementResolverHelper.php

```diff
@@ -58,11 +53,26 @@ public function convertResolveVariantProducts(?ProductCollection $products, Sale
         }));
     }
 
+    public function reloadProducts(ProductCollection $products, SalesChannelContext $context): ProductCollection
+    {
+        if ($products->count() === 0) {
+            return $products;
+        }
+
+        $criteria = new Criteria(array_values(
+            $products->map(fn(ProductEntity $product) => $product->getId())
+        ));
+        $criteria->addAssociations(['options', 'manufacturer', 'media']);
+
+        return $this->salesChannelProductRepository->search($criteria, $context)
+            ->getEntities();
+    }
+
     private function getChildProducts(string $productId, SalesChannelContext $context): EntitySearchResult
     {
         $criteria = new Criteria();
         $criteria->addFilter(new EqualsFilter('parentId', $productId));
-        $criteria->addAssociations(['options', 'manufacturer', 'media']);
+        $criteria->addAssociations(['options', 'manufacturer', 'media', 'cover']);
         return $this->salesChannelProductRepository->search($criteria, $context);
```

Comment:
Da du nur das erste Produkt nutzt, solltest du die Funktion `getChildProducts` in `getFistChildProduct` ändern und direkt das erste Child oder null zurückgeben.

Reply:
hab ich angepasst